File: doc/blog/thread/sql.rb
# sql.rb
require 'sqlite3'
class Main
  def main ()
    db = SQLite3::Database.new(':memory:');
    thread = Thread.new {
      x = db.execute('SELECT 2 * 3')[0][0]
      puts(x)
    }
    x = db.execute('SELECT 4 * 5')[0][0];
    puts(x)
    thread.join
    return 0
  end
end
Main.new.main()

File: doc/blog/thread/sql.mjs
// sql.mjs
import util from 'util';
import sqlite3 from 'sqlite3';
const db = new sqlite3.Database(':memory:');
const execute = util.promisify(db.all.bind(db));
const main = () => {
  execute('SELECT 2 * 3 as x').then(([{x}]) => {
    console.log(x);
  });
  execute('SELECT 4 * 5 as x').then(([{x}]) => {
    console.log(x);
  });
  return 0;
}
main();
